airbyte-server/src/test/java/io/airbyte/server/handlers/ConnectionsHandlerTest.java

```diff
@@ -612,6 +619,321 @@ void testEnumConversion() {
+              )
+          );
+      final AirbyteCatalog catalog2 = new AirbyteCatalog()
+          .streams(
+              List.of(
+                  getStreamAndConfig("stream1", streamConfiguration1),
+                  getStreamAndConfig("stream2", streamConfiguration2)
+              )
+          );
+
+      assertTrue(connectionsHandler.getConfigurationDiff(catalog1, catalog2).isEmpty());
+    }
+
+    @Test
+    void testNoDiffIfStreamAdded() {
+      final AirbyteStreamConfiguration streamConfiguration1 = getStreamConfiguration(List.of("cursor1"), List.of(List.of("pk1")),
+          SyncMode.INCREMENTAL,
+          DestinationSyncMode.APPEND_DEDUP);
+
+      final AirbyteStreamConfiguration streamConfiguration2 = getStreamConfiguration(List.of("cursor2"), List.of(List.of("pk2")),
+          SyncMode.FULL_REFRESH,
+          DestinationSyncMode.OVERWRITE);
+
+      final AirbyteCatalog catalog1 = new AirbyteCatalog()
+          .streams(
+              List.of(
+                  getStreamAndConfig("stream1", streamConfiguration1)
+              )
+          );
+      final AirbyteCatalog catalog2 = new AirbyteCatalog()
+          .streams(
+              List.of(
+                  getStreamAndConfig("stream1", streamConfiguration1),
+                  getStreamAndConfig("stream2", streamConfiguration2)
+              )
+          );
+
+      assertTrue(connectionsHandler.getConfigurationDiff(catalog1, catalog2).isEmpty());
+    }
+
+    @Test
+    void testCursorOrderDoesntMatter() {
+      final AirbyteStreamConfiguration streamConfiguration1 = getStreamConfiguration(List.of("cursor1", "anotherCursor"), List.of(List.of("pk1")), SyncMode.INCREMENTAL,
+          DestinationSyncMode.APPEND_DEDUP);
+
+      final AirbyteStreamConfiguration streamConfiguration1WithOtherCursorOrder = getStreamConfiguration(List.of("anotherCursor", "cursor1"),
+          List.of(List.of("pk1")),
+          SyncMode.INCREMENTAL,
+          DestinationSyncMode.APPEND_DEDUP);
+
+      final AirbyteStreamConfiguration streamConfiguration2 = getStreamConfiguration(List.of("cursor2"), List.of(List.of("pk2")),
+          SyncMode.FULL_REFRESH,
+          DestinationSyncMode.OVERWRITE);
+
+      final AirbyteCatalog catalog1 = new AirbyteCatalog()
+          .streams(
+              List.of(
+                  getStreamAndConfig("stream1", streamConfiguration1),
+                  getStreamAndConfig("stream2", streamConfiguration2)
+              )
+          );
+      final AirbyteCatalog catalog2 = new AirbyteCatalog()
+          .streams(
+              List.of(
+                  getStreamAndConfig("stream1", streamConfiguration1WithOtherCursorOrder),
+                  getStreamAndConfig("stream2", streamConfiguration2)
+              )
+          );
+
+      assertTrue(connectionsHandler.getConfigurationDiff(catalog1, catalog2).isEmpty());
+    }
+
+    @Test
+    void testPkOrderDoesntMatter() {
```

Comment:
Similar question as the above here

Reply:
The order doesn't matter, for example:
- [[a, b, c]] represent 1 pk that contains a tuple made by column a, b and c as a primary key so it should be the same than [[c, b, a]]
- I don't really know if [[a], [b, c]] is the same than [[b, c], [a]] I don't understand how you can define multiple primary keys.

I have enforce the order to use the rule with the more constraint.